modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DefaultArchiveDeployer.java

```diff
@@ -336,8 +336,9 @@ private void undeployArtifact(T artifact, boolean removeData) {
       artifactArchiveInstaller.uninstallArtifact(artifact.getArtifactName());
       if (removeData) {
```

Comment:
Should you just change the logic when this kind of redeploy happens to call this method with removeData=false?